integration-test/src/test/java/uk/gov/gchq/gaffer/integration/junit/extensions/IntegrationTestSuiteExtensionTest.java

```diff
@@ -0,0 +1,353 @@
+                .haveExactly(1, event(uniqueIdSubstring(IntegrationTestSuiteExtension.class.getName()),
+                        finishedWithFailure(message("The initClass @ConfigurationParameter has not been set"))));
+    }
+
+    @Suite
+    @SelectClasses(value = {InitClassDoesNotImplementIntegrationTestSuiteIT.class})
+    static class InitClassDoesNotImplementIntegrationTestSuiteSuite {
+    }
+
+    @ExtendWith(IntegrationTestSuiteExtension.class)
+    static class InitClassDoesNotImplementIntegrationTestSuiteIT {
+        @Test
+        void shouldFail() {
+            fail(ERROR_THE_TEST_SHOULD_NOT_RUN);
+        }
+    }
+
+    @Test
+    void shouldFailInitClassDoesNotImplementIntegrationTestSuite() {
+        EngineTestKit.engine(JUNIT_PLATFORM_SUITE)
+                .selectors(selectClass(InitClassDoesNotImplementIntegrationTestSuiteSuite.class))
+                .configurationParameter(INIT_CLASS, InitClassDoesNotImplementIntegrationTestSuiteSuite.class.getName())
+                .execute()
+                .allEvents()
+                .assertThatEvents()
+                .haveExactly(1, event(uniqueIdSubstring(IntegrationTestSuiteExtension.class.getName()),
+                        finishedWithFailure(message(String.format("The object was not of required type: [%s]. Actual object type: [%s]",
+                                IntegrationTestSuite.class.getName(), InitClassDoesNotImplementIntegrationTestSuiteSuite.class.getName())))));
+    }
+
+    @SelectClasses(value = {TestSkippedIT.class})
+    static class TestSkippedTestSuite extends AbstractTestSuite {
+        TestSkippedTestSuite() {
+            setSchema(TEST_SCHEMA);
+            setStoreProperties(TEST_STORE_PROPERTIES);
+            setTestsToSkip(Collections.singletonMap("shouldSkip", "skipped test"));
+        }
+    }
+
+    @ExtendWith(IntegrationTestSuiteExtension.class)
+    static class TestSkippedIT {
+        @Test
+        void shouldSkip() {
+            fail("The test should be skipped");
+        }
+    }
+
+    @Test
+    void shouldSucceedTestSkippedTest() {
+        EngineTestKit.engine(JUNIT_PLATFORM_SUITE)
+                .selectors(selectClass(TestSkippedTestSuite.class))
+                .configurationParameter(INIT_CLASS, TestSkippedTestSuite.class.getName())
+                .execute()
+                .testEvents()
+                .skipped()
+                .assertThatEvents()
+                .haveExactly(1, event(test(TestSkippedIT.class.getName()), skippedWithReason("skipped test")))
+                .hasSize(1);
+    }
+
+    static class CallCacheTestSuite extends AbstractTestSuite {
+
+        static int instantiated = 0;
+
+        CallCacheTestSuite() {
+            instantiated++;
+            setSchema(TEST_SCHEMA);
+            setStoreProperties(TEST_STORE_PROPERTIES);
+        }
+    }
+
+    @Test
+    void shouldSucceedCallCache(@Mock final ExtensionContext mockExtensionContext) {
+        /* setup */
```

Comment:
Normally we'd used `//` for inline comment. Very minor point.

Reply:
I put the main stages in `/* ... */` comments so they stand out from the other `//` comments in the code. I've found that if you have a lot of comments then they can get lost